File: apple/ListenToWikipedia/Audio/ScaleType.swift
import Foundation

/// The number of notes in a scale per octave.
enum ScaleType: Int, CaseIterable, Identifiable, CustomStringConvertible {
  case pentatonic = 5
  case heptatonic = 7

  var id: Int { rawValue }

  var description: String {
    switch self {
    case .pentatonic: return "Pentatonic (5-note)"
    case .heptatonic: return "Heptatonic (7-note)"
    }
  }
}

File: android/app/src/main/java/me/bryanoltman/listentowikipedia/audio/ScaleType.kt
package me.bryanoltman.listentowikipedia.audio

enum class ScaleType(val noteCount: Int, val displayName: String) {
    PENTATONIC(5, "Pentatonic (5-note)"),
    HEPTATONIC(7, "Heptatonic (7-note)")
}
